agents/craftassist/craftassist_agent.py

```diff
@@ -225,17 +228,38 @@ def init_controller(self):
 
     def perceive(self, force=False):
         """Whenever some blocks are changed, that area will be put into a 
-        buffer which will be force-perceived by the agent in the next step
+        buffer which will be force-perceived by the agent in the next step.
 
         Here the agent first clusters areas that are overlapping on the buffer,
-        then run through all perception modules to perceive
-        and finally clear the buffer when perception is done.
+        then run through all perception modules to perceive and finally clears the
+        buffer when perception is done.
+
+        The agent sends all perception updates to memory in order for them to
+        update the memory state.
         """
         self.areas_to_perceive = cluster_areas(self.areas_to_perceive)
```

Comment:
can you move this down to line 245?

Reply:
sorry I don't understand, self.areas_to_perceive is updated between lines 241 and 244?  if so, can we have whatever update be done in the agent? like returned in the perception_output.  if not, yes, just move to be right before 245 so all mentions are in the same place